src/app/pages/institutional-partnership-application/institutional-partnership-application.js

```diff
@@ -1,82 +1,118 @@
-import componentType, {canonicalLinkMixin} from '~/helpers/controller/init-mixin';
-import TabGroup from '~/components/tab-group/tab-group';
-import ContentGroup from '~/components/content-group/content-group';
-import ProgramDetails from './program-details';
-import Application from './application';
-import {description as template} from './institutional-partnership-application.html';
-import css from './institutional-partnership-application.css';
+import {pageWrapper} from '~/controllers/jsx-wrapper';
+import React, {useState} from 'react';
+import {LoaderPage, RawHTML} from '~/components/jsx-helpers/jsx-helpers.jsx';
+import TabGroup from '~/components/tab-group/tab-group.jsx';
+import ContentGroup from '~/components/content-group/content-group.jsx';
+import './institutional-partnership-application.css';
 
-const spec = {
-    template,
-    css,
-    view: {
-        tag: 'main',
-        classes: ['institutional-page', 'page']
-    },
-    model() {
-        return this.pageData && {
-            heading: this.pageData.heading,
-            headingYear: this.pageData.heading_year,
-            testimonial: this.testimonials[this.selectedTab]
-        };
-    },
-    slug: 'pages/institutional-partnership'
-};
-const BaseClass = componentType(spec, canonicalLinkMixin);
+const labels = ['Program details', 'Application'];
+
+function TestimonialBlock({selectedLabel, data}) {
+    const testimonials = {
+        'Program details': {
+            block: data.quote,
+            name: data.quoteAuthor,
+            address1: data.quoteTitle,
+            address2: data.quoteSchool
+        },
+        'Application': {
+            block: data.applicationQuote,
+            name: data.applicationQuoteAuthor,
+            address1: data.applicationQuoteTitle,
+            address2: data.applicationQuoteSchool
+        }
+    };
+    const testimonial = testimonials[selectedLabel];
+
+    return (
+        <div class="testimonial">
+            <div class="boxed">
+                <div class="testimonial-box">
+                    <p class="text-block">{testimonial.block}</p>
+                    <div class="writer-info">
+                        <p>
+                            <strong>-{testimonial.name}</strong><br />
+                            {testimonial.address1}<br />
+                            {testimonial.address2}
+                        </p>
+                    </div>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+function Application() {
+    return (
+        <div className="application">
+            <iframe
+                class="form-iframe"
+                src="https://docs.google.com/forms/d/e/1FAIpQLSfI0XpCC-4ag0u9vHizT5w2JLkwcv39HVTZnvQHEtmQj8PJwQ/viewform"
+            />
+        </div>
+    );
+}
 
-export default class Institutional extends BaseClass {
+function ProgramDetails({model}) {
+    return (
+        model[0].map((section) =>
+            <div key={section.heading}>
```

Comment:
will the headings always be unique?  Something I've learned with react is that having duplicate keys can cause really weird re-rendering.  Luckily it'll complain about that as well as missing ones, but not sure what preact does

Reply:
I get red error messages in the console for duplicate keys. In this case, headings really should be unique.